packages/notte-browser/src/notte_browser/window.py

```
@@ -335,30 +383,64 @@ async def screenshot(self, retries: int = config.empty_page_max_retry) -> bytes:
+                for attempt in range(1, self.CDP_SCREENSHOT_MAX_ATTEMPTS + 1):
+                    attempt_start = time.monotonic()
+                    try:
+                        return await self._cdp_screenshot()
+                    except asyncio.CancelledError:
+                        raise
+                    except asyncio.TimeoutError:
+                        attempt_ms = (time.monotonic() - attempt_start) * 1000
+                        logger.warning(
+                            f"CDP screenshot timed out after {attempt_ms:.0f}ms for {self.page.url} (attempt {attempt}/{self.CDP_SCREENSHOT_MAX_ATTEMPTS}, budget={self.CDP_SCREENSHOT_TIMEOUT_S}s)"
+                        )
+                    except Exception as e:
```

Ruff (0.15.11)

[warning] Do not catch blind exception: `Exception`

(BLE001)